### PR TITLE
A tiny Js world - OOP

### PR DESCRIPTION
**Pull request by olek557**
### A tiny Js world - OOP

[Demo](https://olek557.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/olek557/a-tiny-JS-world/tree/populate-world-oop)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by OleksiyRudenko** [comment]
@olek557 Thank you for your submission.

Please check yourself against the list below:
 * [ ] Implement a base class to inherit from
 * [ ] Employ default parameters
 * [ ] Each species is represented with its own class
 * [ ] No need to specify species at instantiation
 * [ ] Classes for species that do not have hands by natural design
       do not consequently have `hands` or any equivalent property
 * [ ] All inhabitants are stored in a container (array or object)
 * [ ] JS native features are intensively employed (`const`, `let`, `Array.map|join|forEach|...`, etc)
 * [ ] [OOP, SOLID and DRY](https://github.com/OleksiyRudenko/a-tiny-JS-world/blob/master/README.md#learn-on-your-own)
       principles are intensively employed
 
Optional level up (not required to implement):
 * [ ] Friends list is a list of objects refs rather than names (strings)
 * [ ] Cat-woman class is built employing composition rather than inheritance only
  
[ES6 classes cheat-sheet](https://gist.github.com/OleksiyRudenko/672d39b08d9d0da4e179aca49876c58b)

Please, confirm all items above are positive. Please, ask a question if anything above requires clarification.

**Comment by olek557** [comment]
@OleksiyRudenko Rewrote task in accordance to provided list. Please review.

File: submissions/olek557/js-post-oop-task/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+class Inhabitant {
+  constructor(name, gender, legs, phrase) {
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.phrase = phrase;
+  }
+  fullInfo() {
+      return `name: ${this.name}, gender: ${this.gender}, legs: ${this.legs}, say: ${this.phrase}`;
+  }
```
Comment (OleksiyRudenko) [inline submissions/olek557/js-post-oop-task/index.js]: Let's make code DRYer.
Hint: ``[varA, "text", `some text: ${variable}`].join('delimiter')``

Bonus: Also rename this method (and in a child class as well) as `toString`. This way your method becomes magic. Explained later.

File: submissions/olek557/js-post-oop-task/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// ======== OBJECTS DEFINITIONS ========
+class Inhabitant {
+  constructor(name, gender, legs, phrase) {
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.phrase = phrase;
+  }
```
Comment (OleksiyRudenko) [inline submissions/olek557/js-post-oop-task/index.js]: Tech spec requires `species`. Please, add the feature.

File: submissions/olek557/js-post-oop-task/index.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+let inhabitants = [
+  new Dog('Barsik', 'male'),
+  new Cat('Max', 'male'),
+  new Human('Alex', 'female', "Hello world")
+];
+
+// ======== OUTPUT ========
+inhabitants.forEach(inhabitant => {
+  print(inhabitant.fullInfo())
+});
```
Comment (OleksiyRudenko) [inline submissions/olek557/js-post-oop-task/index.js]: When you have your string formation method named `toStrting` you can just `print(inhabitant)`.
`toString()` will be called implicitly.

File: submissions/olek557/js-post-oop-task/index.js (hunk 1 above)
Comment (olek557) [reply submissions/olek557/js-post-oop-task/index.js]: 👍

File: submissions/olek557/js-post-oop-task/index.js (hunk 3 above)
Comment (olek557) [reply submissions/olek557/js-post-oop-task/index.js]: Look's cool!! thanks!

**Comment by olek557** [comment]
@OleksiyRudenko Thanks for comments, i changed this things.

File: submissions/olek557/js-post-oop-task/index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+// ======== OBJECTS DEFINITIONS ========
+class Inhabitant {
+  constructor(name, gender, legs, phrase, species) {
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.phrase = phrase;
+    this.species = species;
+  }
```
Comment (OleksiyRudenko) [inline submissions/olek557/js-post-oop-task/index.js]: Here is your another bonus:
In this particular case you may enjoy `this.constructor.name` which is a magic property,

**Pull request merged**